lib/moleculer/packets/discover.rb

```diff
@@ -5,6 +5,20 @@ module Packets
     ##
     # Represents a DISCOVER packet
     class Discover < Base
+      def initialize(config, data)
+        super(config, data)
+
+        node     = HashUtil.fetch(data, :node, nil)
+        @node_id = HashUtil.fetch(data, :node_id, node)
+      end
+
+      def topic
```

Comment:
Packets::Discover && Packets::Info have this logic in it. Would it be better to move the topic method into the Base class?

Reply:
I'm not quite sure I understand. The base class already has a topic method, the packets need to override it as necessary. The discover and info packets in particular require logic because they can either come from a broadcast method, or come from a direct source.